show/bgp_common.py

```diff
@@ -0,0 +1,420 @@
+    # If other protocol are also single nexthop not in this list just add them
+    single_nh_proto = ["connected"]
+    new_info_l = copy.deepcopy(r_info_l)
+    # check if this route already exists. if so, combine nethops and update the count
+    if route in combined_route:
+        while len(new_info_l):
+            new_info = new_info_l.pop()
+            proto_matched = False
+            skip_this_new_info = False
+            for j in range(0, len(combined_route[route])):
+                if new_info['protocol'] == combined_route[route][j]['protocol']:
+                    proto_matched = True
+                    if new_info['protocol'] in single_nh_proto:
+                        # For single nexhop protocols handling differs where it should be either not added or add ASIS
+                        # If this is new, need to add the new_info. else skip this new_info
+                        if combined_route[route][j]['nexthops'][0]['interfaceName'] == new_info['nexthops'][0]['interfaceName']:
+                            skip_this_new_info = True
+                            break
+                    else:
+                        # protocol may contain multiple nexthops. Need to filter by nexthops
+                        additional_nh_l = []
+                        while len(new_info['nexthops']):
+                            nh = new_info['nexthops'].pop()
+                            found = False
+                            for y in range(0, len(combined_route[route][j]['nexthops'])):
+                                if "interfaceName" in nh and "interfaceName" in combined_route[route][j]['nexthops'][y]:
+                                    if nh['interfaceName'] == combined_route[route][j]['nexthops'][y]['interfaceName']:
+                                        found = True
+                                        break
+                                elif "active" not in nh and "active" not in combined_route[route][j]['nexthops'][y]:
+                                    if nh['ip'] == combined_route[route][j]['nexthops'][y]['ip']:
+                                        found = True
+                                        break
+                            if not found:
+                                additional_nh_l.append(copy.deepcopy(nh))
+
+                        if len(additional_nh_l) > 0:
+                            combined_route[route][j]['internalNextHopNum'] + len(additional_nh_l)
+                            if combined_route[route][j]['internalNextHopActiveNum'] > 0 and new_info['internalNextHopActiveNum'] > 0:
+                                combined_route[route][j]['internalNextHopActiveNum'] + len(additional_nh_l)
+                            while len(additional_nh_l):
+                                a_nh = additional_nh_l.pop()
+                                combined_route[route][j]['nexthops'].append(a_nh)
```

Comment:
use `+` operator here ?

Reply:
@arlakshm Agreed. will replace the while loop with + operator to combine the nexthop list.